monai/transforms/croppad/dictionary.py

```diff
@@ -1525,6 +1529,53 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, N
         return d
 
 
+class PatchIterd(MapTransform):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.PatchIter`.
+    Return a patch generator for dictionary data and the coordinate, Typically used
+    with :py:class:`monai.data.GridPatchDataset`.
+    Suppose all the expected fields specified by `keys` have same shape.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: monai.transforms.MapTransform
+        patch_size: size of patches to generate slices for, 0/None selects whole dimension
+        start_pos: starting position in the array, default is 0 for each dimension
+        mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
+            ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
+            One of the listed string values or a user supplied function. Defaults to ``"wrap"``.
+            See also: https://numpy.org/doc/1.18/reference/generated/numpy.pad.html
+        allow_missing_keys: don't raise exception if key is missing.
+        kwargs: other arguments for the `np.pad` function.
+            note that `np.pad` treats channel dimension as the first dimension.
+
+    """
+
+    backend = PatchIter.backend
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        patch_size: Sequence[int],
+        start_pos: Sequence[int] = (),
+        mode: Union[NumpyPadMode, str] = NumpyPadMode.WRAP,
+        allow_missing_keys: bool = False,
+        **kwargs,
+    ):
+        super().__init__(keys, allow_missing_keys)
+        self.patch_iter = PatchIter(patch_size=patch_size, start_pos=start_pos, mode=mode, **kwargs)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]):
+        d = dict(data)
+        for t in zip(*[self.patch_iter(d[key]) for key in self.key_iterator(d)]):
+            coords = t[0][1]  # use the coordinate of the first item
+            ret = {k: v[0] for k, v in zip(self.key_iterator(d), t)}
+            # fill in the extra keys with unmodified data
+            for key in set(d.keys()).difference(set(self.keys)):
+                ret[key] = deepcopy(d[key])
+            yield ret, coords
```

Comment:
would be great to also store the coords as part of the data dict.

Reply:
OK, I added the `coords` in the dict in the latest commit.

Thanks.